### PR TITLE
Ninth task add category entity and tied classes

### PR DESCRIPTION
**Pull request by apanyutin**

### CODE REVIEW

File: src/main/java/com/example/controller/AuthController.java [hunk 1]
```diff
@@ -19,11 +22,18 @@
...
     @Operation(summary = "User registration", description = "New user registration")
     @PostMapping("/registration")
     public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto requestDto)
             throws RegistrationException {
         return userService.register(requestDto);
     }
+
+    @Operation(summary = "User login", description = "New user registration")
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/controller/AuthController.java]: Why these changes in Category PR ?

File: src/main/java/com/example/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -63,4 +64,10 @@ public List<BookDto> search(BookSearchParameters params, Pageable pageable) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/service/impl/BookServiceImpl.java:67]: Do we need to setCategories during updating and saving book?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/example/controller/AuthController.java (hunk 1 above)
Comment (apanyutin) [reply src/main/java/com/example/controller/AuthController.java]: I don't remember. Maybe I added @Valid annotation

File: src/main/java/com/example/service/impl/BookServiceImpl.java (hunk 2 above)
Comment (apanyutin) [reply src/main/java/com/example/service/impl/BookServiceImpl.java:67]: Yes. Now it's working.

File: src/main/java/com/example/security/AuthService.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthService {
+    private final JwtUtil jwtUtil;
+    private final AuthenticationManager authenticationManager;
+
+    public UserLoginResponseDto authenticate(@Valid UserLoginRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/security/AuthService.java]:
```suggestion
    public UserLoginResponseDto authenticate(UserLoginRequestDto requestDto) {
```

File: src/main/java/com/example/service/impl/BookServiceImpl.java (hunk 2 above)
Comment (Elena-Bruyako) [reply src/main/java/com/example/service/impl/BookServiceImpl.java:67]: Didn't find these logic.

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
See comments

File: src/main/java/com/example/service/impl/BookServiceImpl.java (hunk 2 above)
Comment (apanyutin) [reply src/main/java/com/example/service/impl/BookServiceImpl.java:67]: I implemented it in BookMapper interface. with help two methods:
@AfterMapping
    default void setCategoryIds(@MappingTarget BookDto bookDto, Book book) {
        bookDto.setCategoryIds(book.getCategories().stream()
                .map(Category::getId)
                .collect(Collectors.toSet()));
    }

    @AfterMapping
    default void setCategoryIds(@MappingTarget Book book, CreateBookRequestDto requestDto) {
        book.setCategories(requestDto.getCategoryIds().stream()
                .map(Category::new)
                .collect(Collectors.toSet()));
    }

**Review by Elena-Bruyako — APPROVED** [review]
well done
